core/client/app/controllers/settings/general.js

```diff
@@ -76,6 +76,20 @@ export default Controller.extend(SettingsSaveMixin, {
         }
     }),
 
+    facebookUrl: computed('model.facebook', function () {
+        return this.get('model.facebook');
+    }),
+
+    twitterUrl: computed('model.twitter', function () {
```

Comment:
Yeah, that's definitely an option! 😄 That way you could have a "twitter" object with both a `url` property and a `username` CP that handles conversion to/from the url as well as validation.